stdlib/LinearAlgebra/src/cholesky.jl

```diff
@@ -835,3 +835,33 @@ then `CC = cholesky(C.U'C.U - v*v')` but the computation of `CC` only uses
 `O(n^2)` operations.
 """
 lowrankdowndate(C::Cholesky, v::AbstractVector) = lowrankdowndate!(copy(C), copy(v))
+
+function diag(C::Cholesky{T}, k::Int = 0) where {T}
+    N = size(C, 1)
+    absk = abs(k)
+    z = Vector{T}(undef, N - absk)
+    UL = C.factors
+    if C.uplo == 'U'
+        for i in 1:N - abs(k)
```

Comment:
Nitpick:
```suggestion
        for i in 1:(N - absk)
```
And similarly in the for loop in the other branch.